clients/elrond/transactionHandler.go

```diff
@@ -18,10 +18,14 @@ type transactionHandler struct {
 	nonceTxHandler          NonceTransactionsHandler
 	relayerPrivateKey       crypto.PrivateKey
 	singleSigner            crypto.SingleSigner
+	roleProvider            roleProvider
 }
 
 // SendTransactionReturnHash will try to assemble a transaction, sign it, send it and, if everything is OK, returns the transaction's hash
 func (txHandler *transactionHandler) SendTransactionReturnHash(ctx context.Context, builder builders.TxDataBuilder, gasLimit uint64) (string, error) {
+	if !txHandler.roleProvider.IsWhitelisted(txHandler.relayerAddress) {
+		return "", errRelayerNotWhitelisted
```

Comment:
this will essentially make the non-whitelisted relayers to output an error and return to the get pending batch step.